Tests/CatalogItems/New-RsDataSource.Tests.ps1

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) 2016 Microsoft Corporation. All Rights Reserved.
+# Licensed under the MIT License (MIT)
+
+Function Create-PSCredential () 
+{
+param(
+        [Parameter(Mandatory = $True)]
+        [string]$User,
+        [Parameter(Mandatory = $True)]
+        [string]$Password
+    )
+       $securePassword = ConvertTo-SecureString $Password -AsPlainText -Force
+       $ps_credential = New-Object System.Management.Automation.PSCredential ($User, $securePassword)
+       Return $ps_redential 
+}
+
+Function Get-ExistingExtension () 
+{
+        $proxy = New-RsWebServiceProxy
+        return $proxy.ListExtensions("Data")[0].Name
+}
+
+Describe "New-RsDataSource" {
+    Context "Create RsDataSource with minimun parameters"{
+        # Declare datasource Name, Extension, CredentialRetrieval, and DataSource path.
+        $dataSourceName = 'SutDataSourceMinParameters' + [guid]::NewGuid()
+        $extension = Get-ExistingExtension
+        $credentialRetrieval = 'None'
+        $dataSourcePath = '/' + $dataSourceName
+        New-RsDataSource -RsFolder '/' -Name $dataSourceName -Extension $extension -CredentialRetrieval $credentialRetrieval
+        It "Should be a new data source" {
+            {Get-RsDataSource -Path $dataSourcePath } | Should not throw
```

Comment:
I would recommend you to also check extension and credential retrieval properties are correct.